src/Features/LanguageServer/Protocol/Handler/Diagnostics/DiagnosticSourceProviders/IDiagnosticSourceManager.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler.Diagnostics.DiagnosticSources;
+
+/// <summary>
+/// Manages the diagnostic sources that provide diagnostics for the language server.
+/// </summary>
+internal interface IDiagnosticSourceManager
+{
+    /// <summary>
+    /// Returns the names of all the sources that provide diagnostics for the given <paramref name="isDocument"/>.
+    /// </summary>
+    /// <param name="isDocument"><see langword="true" /> for document sources and <see langword="false" /> for workspace sources.</param>
+    ImmutableArray<string> GetSourceNames(bool isDocument);
```

Comment:
is this more accurate as GetSourceProviderNames?  Also, given the CreateDocumentXXX and CreateWorkspaceXXX below, should this be GetDocumetnSourceProviderNames and GetWorkspaceSourceProviderNames?